src/NuGet.Services.AzureSearch/SearchService/SearchResponseBuilder.cs

```diff
@@ -323,16 +323,27 @@ private static string TitleThenId(IBaseMetadataDocument document)
             }
             else
             {
+                var resultData = results.Select(x =>
+                {
+                    var package = toPackage(x.Document);
+                    package.Debug = request.ShowDebug ? x : null;
+                    return package;
+                });
+
+                // We sort here to make sure to sort on the same data displayed to the user (i.e: to avoid having inconsistencies in the displayed order of the packages)
```

Comment:
I would mention that the real sorting is done by Azure Search. However, the downloads data we sort on (the Azure Search index's downloads) is different from the downloads data we display (the downloads auxiliary file), so we do another round of sorting here to remove inconsistencies. I would also mention that the other sort orders don't have this problem and don't need sorting here.